### PR TITLE
Test DLS and FLS against runtime fields

### PR DESCRIPTION
**Pull request by javanna**
This PR introduces integration tests for Document level security with queries against runtime fields, as well as field level security used in combination with runtime fields.

Relates to #59332

### CODE REVIEW

File: x-pack/plugin/runtime-fields/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    public void testFLSOnRuntimeField() throws IOException {
+        Request createIndex = new Request("PUT", "/fls");
+        createIndex.setJsonEntity("{\n" +
+            "    \"mappings\" : {\n" +
+            "        \"properties\" : {\n" +
+            "            \"date\" : {\"type\" : \"keyword\"},\n" +
+            "            \"year\" : {\n" +
+            "              \"type\" : \"runtime_script\", \n" +
+            "              \"runtime_type\" : \"keyword\",\n" +
+            "              \"script\" : \"emitValue(doc[\\\"date\\\"].value.substring(0,4))\"\n" +
```
Comment (nik9000) [inline x-pack/plugin/runtime-fields/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java:178]: s/\\\\"/'` I think will work.

File: x-pack/plugin/runtime-fields/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        assertOK(adminClient().performRequest(indexDoc1));
+
+        Request indexDoc2 = new Request("PUT", "/fls/_doc/2");
+        indexDoc2.setJsonEntity("{\n" +
+            "    \"date\" : \"2016-11-15T14:12:12\"\n" +
+            "}\n");
+        assertOK(adminClient().performRequest(indexDoc2));
+
+        Request indexDoc3 = new Request("PUT", "/fls/_doc/3");
+        indexDoc3.addParameter("refresh", "wait_for");
```
Comment (nik9000) [inline x-pack/plugin/runtime-fields/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java:198]: I don't think we need to wait_for a refresh, we can just do `true`, I think.

Comment (javanna) [reply x-pack/plugin/runtime-fields/qa/with-security/src/test/java/org/elasticsearch/xpack/security/PermissionsIT.java:198]: both work I guess

**Comment by javanna** [comment]
run elasticsearch-ci/packaging-sample-windows

**Pull request merged**